db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLDate.java

```diff
@@ -79,6 +82,13 @@
 	 +     * The JodaTime has problems with all the years before 1884
 	 +     */
 
+	public static final ThreadLocal<GregorianCalendar> GREGORIAN_CALENDAR =
+			new ThreadLocal<GregorianCalendar>() {
+				@Override
+				protected GregorianCalendar initialValue() {
+					return new GregorianCalendar();
+				}
```

Comment:
I think we should .clear() the instance in ThreadLocal.get(), otherwise some fields could be polluted from previous usages.